CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKIconTextureAtlas.java

```diff
@@ -219,16 +225,45 @@ public void IncrementDescriptorTypeRequirements(CVKDescriptorPoolRequirements re
     
     @Override
     public int DisplayUpdate() {
-        Destroy();
-        return CreateAtlas();
+        int ret = VK_SUCCESS;
+        
+        if (needsSaveToFile) {
+            ret = SaveToFile(fileToSave);
+            needsSaveToFile = false;
+        }
+        
+        if (needsAtlasRebuilt) {
+            Destroy();
+            ret = CreateAtlas();
+            needsAtlasRebuilt = false;
+        }
+        
+        return ret;
     }
     
     @Override
     public boolean NeedsDisplayUpdate() {
-        return loadedIcons.size() > lastTransferedIconCount;
+       
+        needsAtlasRebuilt = (loadedIcons.size() > lastTransferedIconCount);
+        
+        return needsSaveToFile || needsAtlasRebuilt;
     }  
     
     
+    // ========================> Tasks <======================== \\
+    
+    public CVKRenderUpdateTask TaskSaveToFile(File file) {
+        //=== EXECUTED BY CALLING THREAD (VisualProcessor) ===//
+        needsSaveToFile = true;
+        fileToSave = file;
```

Comment:
Maybe move to the lambda if it lets you.

Reply:
moved